### DISCUSSION TITLE
Feature Request: Implement RBAC to restrict users using some models

### DISCUSSION THREAD

**Discussion by dominpm** [open-webui/open-webui#5853] [General]
**Is your feature request related to a problem? Please describe.**
It would be a really nice feature for the admins to have to restrict the use of certain models to certain users. Meaning that there would be different roles, where different roles give access to different models/features.

**Describe the solution you'd like**
Once a user registers in the app, admin would have the ability to set them in different groups/roles. Those roles will restrict which models can they use and which not.

**Extra**
It would be awesome too to integrate this with Azure Directory to allow users register (which is currently allowed); but their groups are inherited in OpenWebUI so that RBAC is easier to manage.

**Comment by doxav** [comment]
It should also apply to pipelines and RAG collections

**Reply by dominpm** [reply]
Yes, and it would also help restrict how different users have permissions in the UI to edit custom settings.

**Comment by 8thHalfHour** [comment]
![plus-one_cool](https://github.com/user-attachments/assets/0cc78f2f-5922-4c9e-a295-52e9b5793061)
YES, please. Restricting access to specific functions, workflows, and custom AIs to only those who need it could really optimize costs and help us look more closely at usage metrics. Plus, aligning the UI and settings with available options in a business environment means user expectations = the capabilities of each model or tool (this has been a frustration in my instance, sometimes causing issues across the platform when models don't support that function). Also, managing user access to resources matches everything else admins are expected to do in a healthy compliant system anyway, right, so this would keep our Open WebUI deployment consistent with the rest of our policies and processes. Makes for happy admins! Bonus points for more control = simplifying future modifications and maintenance. Great request! I need it yesterday. LOL

**Comment by jkl0ckenkemper** [comment]
**As an** Administrator managing an internal open-web-ui multi user system 
**I want** to be able to utilize RBAC roles to control and define user permissions
**So that** I can slice and dice access to models, weights, etc to define workflows more effectively 

---

### Specifics on what I want

- [ ] Criterion 1: Custom Roles (Power user, manager, etc) can be created
- [ ] Criterion 2: Users can be moved between roles levels within the UI
- [ ] Criterion 3: All model configuration item changes can be goverened by user permission
- [ ] Criterion 4: All model utilization selections can be goverened by user permission


---

### Notes

If implimented, I as an Administrator would be able to define different roles for different users.

1) Admin = Same as today
2) Power user = Access to modify model weights, choose from an expanded list of models, manage other lower level users, etc.
3) Base level user = same as today.
